script.js

```diff
@@ -94,168 +122,214 @@ const setupRound = (wordPairs, pairRenderLimitIndex) => {
  * @param className - the class to add and then remove
  */
 function highlightElements(elements, className) {
-  elements.forEach((element) => element.classList.add(className));
+  // Highlighted elements should be disabled during the animation
+  elements.forEach((element) => {
+    element.classList.add(className);
+    element.style.pointerEvents = "none";
```

Comment:
this is not really what I had in mind :D You still have the logic in CSS, you just dynamically add the CSS here. Its better, but not ideal.

What I had in mind was to add `if` condition in the function that reacts to the events (In your case, this function is `checkIfMatch`. Something like

```javascript
const checkIfMatch = (event) => {
  const clickedElement = event.target; 
  // block any action for elements that are already highlighted
  const elementIsBlocked = // ...add your logic - check the classes etc
  if (elementIsBlocked) return;
  /// ...